mypyc/irbuild/builder.py

```diff
@@ -723,6 +726,10 @@ def call_refexpr_with_args(
 
         # Handle data-driven special-cased primitive call ops.
         if callee.fullname is not None and expr.arg_kinds == [ARG_POS] * len(arg_values):
+            call_c_ops_candidates = c_function_ops.get(callee.fullname, [])
+            target = self.builder.matching_call_c(call_c_ops_candidates, arg_values, expr.line)
```

Comment:
We should probably pass the expected type here, similar to `matching_primitive_op` below (`self.node_type(expr)`).